index.d.ts

```diff
@@ -16,20 +18,20 @@ export type Mapper<ValueType, KeyType, MappedValueType> = (
 export default function pProps<
 	KeyType extends unknown,
 	ValueType extends unknown,
-	MappedValueType = ValueType
+	MappedValueType = PromiseResult<ValueType>
 >(
-	input: Map<KeyType, PromiseLike<ValueType> | ValueType>,
+	input: Map<KeyType, ValueType>,
 	mapper?: Mapper<ValueType, KeyType, MappedValueType>,
 	options?: Options
 ): Promise<Map<KeyType, MappedValueType>>;
 export default function pProps<
-	KeyType extends string,
-	ValueType extends unknown,
-	MappedValueType = ValueType
+	InputType extends {[key: string]: any},
```

Comment:
Please change this to:
```suggestion
	InputType extends {[key: string]: unknown},
```